Modules/LearningSequence/classes/class.ilObjLearningSequenceGUI.php

```diff
@@ -57,24 +55,69 @@ public function executeCommand()
 				$this->ctrl->forwardCommand($gui);
 				break;
 
+			case "ilinfoscreengui":
+				$this->info($cmd);
+				break;
+			case "ilpermissiongui":
+				$this->permissions($cmd);
+				break;
+			case "ilobjlearningsequencesettingsgui":
+				$this->settings($cmd);
+				break;
+
+			case "ilobjlearningsequencecontentgui":
 			case false:
+
+				//adjust command
+				if ($cmd === '') {
+					$cmd = self::TAB_VIEW;
+				}
+
+				//do command
 				switch ($cmd) {
-					case "cancelDelete":
-						$cmd = "view";
-					case "create":
-					case "save":
-					case "view":
-					case "cancel":
-					case "edit":
+					case self::TAB_INFO:
+					case 'infoScreen':
+						$this->info();
+						break;
+
+					case self::TAB_CONTENT:
+					case self::TAB_SETTINGS:
+					case self::TAB_VIEW:
 						$this->$cmd();
 						break;
+					default:
+						throw new ilException("ilObjLearningSequenceGUI: Invalid command '$cmd'");
 				}
 				break;
 			default:
```

Comment:
wenn du das 2. switch in den defaultzweig des ersten switch packst können wir auf die 2. throw anweisung verzichten